tests/web/test_web_settings.py

```
@@ -159,10 +170,10 @@ def test_save_general_settings(self, tmp_path: Path, _patch_settings_file: Path)
         assert response.status_code == 200
         assert "General settings saved" in response.text
 
-        # Verify persisted
-        data = json.loads(_patch_settings_file.read_text(encoding="utf-8"))
-        assert data["default_input_dir"] == "/tmp/input"  # noqa: test-hardcoded-paths
-        assert data["default_output_dir"] == "/tmp/output"  # noqa: test-hardcoded-paths
+        # Verify persisted to the shared AppConfig, not web-settings.json.
+        app_config = _load_app_config(tmp_path)
+        assert app_config.default_input_dir == "/tmp/input"  # noqa: test-hardcoded-paths
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp/input"

(S108)